spec/bolt/inventory/inventory2_spec.rb

```diff
@@ -743,6 +1071,123 @@ def common_data(transport)
+        }] }
+      }
+      let(:inventory) { Bolt::Inventory::Inventory2.new(data) }
+      let(:expected_options) {
+        { "connect-timeout" => 10,
+          "tty" => false,
+          "load-config" => true,
+          "disconnect-timeout" => 11,
+          "password" => 'sshpass',
+          "interpreters" => { ".rb" => "/foo/ruby" } }
+      }
+      let(:expected_config) {
+        {
+          "ssh" => {
+            "disconnect-timeout" => 11,
+            "interpreters" => { ".rb" => "/foo/ruby" },
+            "password" => "sshpass"
+          },
+          "transport" => "ssh"
+        }
+      }
+
+      it 'sets config on a target' do
+        target = inventory.get_target('target')
+        expect(target.protocol).to eq('ssh')
+        expect(target.password).to eq('sshpass')
+        expect(target.config).to eq(expected_config)
+        inventory.set_config(target, 'transport', 'local')
+        expect(target.protocol).to eq('local')
+        expect(target.config).to eq(expected_config.merge('transport' => 'local'))
+      end
+
+      it 'overrides a default value not specified in inventory' do
+        target = inventory.get_target('target')
+        expect(target.options).to eq(expected_options)
+        expect(target.config).to eq(expected_config)
+        inventory.set_config(target, %w[ssh tty], true)
+        expected_config['ssh']['tty'] = true
+        expect(target.config).to eq(expected_config)
+      end
+
+      it 'sets a nested config value without disrupting other values' do
+        target = inventory.get_target('target')
+        expect(target.options).to eq(expected_options)
+        expect(target.config).to eq(expected_config)
+        inventory.set_config(target, ['ssh', 'interpreters', '.py'], '/foo/python')
+        expected_options["interpreters"].merge!(".py" => "/foo/python")
+        expected_config['ssh']['interpreters'].merge!(".py" => "/foo/python")
+        expect(target.options).to eq(expected_options)
+        expect(target.config).to eq(expected_config)
+      end
+
+      it 'overrides entire config when an key_or_key_path is an empty string' do
+        target = inventory.get_target('target')
+        expect(target.options).to eq(expected_options)
+        expect(target.config).to eq(expected_config)
+        inventory.set_config(target, '', 'transport' => 'winrm', 'winrm' => { 'password' => 'winrmpass' })
+        expect(target.protocol).to eq('winrm')
+        expect(target.password).to eq('winrmpass')
+        winrm_conf = { "password" => "winrmpass",
+                       "ssl" => true,
+                       "ssl-verify" => true,
+                       "file-protocol" => "winrm",
+                       "connect-timeout" => 10 }
+        expect(target.options).to include(winrm_conf)
+        expected_config['transport'] = 'winrm'
+        expected_config['winrm'] = { 'winrm' => { 'password' => 'winrmpass' } }
+      end
+    end
+  end
+
+  describe 'set_config' do
```

Comment:
I don't follow why there's 2 `describe 'set_config'` here? I think you just need one